internal/deb/extract_test.go

```diff
@@ -352,6 +352,40 @@ var extractTests = []extractTest{{
 		},
 	},
 	error: `cannot extract from package "test-package": path /dir/ requested twice with diverging mode: 0777 != 0000`,
+}, {
+	summary: "Hard link must be created if specified in the tarball",
+	pkgdata: testutil.MustMakeDeb([]testutil.TarEntry{
+		testutil.Dir(0755, "./"),
+		testutil.Reg(0644, "./file1.txt", "text for file1.txt"),
+		testutil.Hln(0644, "./file2.txt", "./file1.txt"),
```

Comment:
Can we change the names so we can see at a glance which one is the link? Maybe something like:
```go
	testutil.Reg(0644, "./file", "whatever"),
	testutil.Hln(0644, "./linked-to-file", "./file"),
```

Reply:
In fact, now that I think about it, there is no test in extractor that tests all types of paths (Reg, Dir, Lnk, Hln), let's make this test that.

Reply:
Fixed in [ffd9951](https://github.com/letFunny/chisel/pull/17/commits/ffd99512f315c63893ae49055f3ad630d6238dd6)

Reply:
Combined test added in [1237a08](https://github.com/letFunny/chisel/pull/17/commits/1237a08b3858690d5c8798992905f5e8f8df3426).